File: 1208_get_equal_substrings_within_budget/equalSubstring.go
func equalSubstring(s string, t string, maxCost int) int {
	n := len(s)

	l, r := 0, 0

	for r = 0; r < n; r++ {
		maxCost -= abs(int(s[r]) - int(t[r]))
		if maxCost < 0 {
			maxCost += abs(int(s[l]) - int(t[l]))
			l++
		}
	}

	return r - l

}

func abs(n int) int {
	if n > 0 {
		return n
	}
	return -n
}

File: 1208_get_equal_substrings_within_budget/equalSubstring.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        n = len(s)

        l = 0

        for r in range(n):
            maxCost -= abs(ord(s[r]) - ord(t[r]))

            if maxCost < 0:
                maxCost += abs(ord(s[l]) - ord(t[l]))
                l += 1

        return r - l + 1